File: opencode/aspnet/rawdog/LooseNotes/Models/ShareLink.cs
using System.ComponentModel.DataAnnotations;

namespace LooseNotes.Models;

public class ShareLink
{
    public int Id { get; set; }
    
    [Required]
    public string Token { get; set; } = string.Empty;
    
    public bool IsActive { get; set; } = true;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime? LastAccessedAt { get; set; }
    
    public int AccessCount { get; set; } = 0;
    
    public int NoteId { get; set; }
    public Note Note { get; set; } = null!;
}

File: claude/jsp/securable/LooseNotes/src/main/java/com/loosenotes/model/ShareLink.java
package com.loosenotes.model;

import java.time.LocalDateTime;

/**
 * Represents a share link for a note.
 * SSEM: Authenticity - token is cryptographically random.
 * SSEM: Confidentiality - token is opaque, not derivable from note ID.
 */
public class ShareLink {

    private long id;
    private long noteId;
    /** Cryptographically random URL-safe token. Never log this value. */
    private String token;
    private LocalDateTime createdAt;
    private boolean active;

    public ShareLink() {
        this.active = true;
    }

    // ---- Getters ----

    public long getId() { return id; }

    public long getNoteId() { return noteId; }

    /** Returns the share token. Handle with care - treat as a credential. */
    public String getToken() { return token; }

    public LocalDateTime getCreatedAt() { return createdAt; }

    public boolean isActive() { return active; }

    // ---- Setters ----

    public void setId(long id) { this.id = id; }

    public void setNoteId(long noteId) { this.noteId = noteId; }

    public void setToken(String token) { this.token = token; }

    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public void setActive(boolean active) { this.active = active; }
}
